base/math.jl

```diff
@@ -166,6 +170,43 @@ function hypot{T<:AbstractFloat}(x::T, y::T)
     x * sqrt(one(r)+r*r)
 end
 
+# 3-arg hypot
+hypot(x::Real, y::Real, z::Real) =
+    hypot(promote(float(x), float(y), float(z))...)
+function hypot{T<:AbstractFloat}(x::T, y::T, z::T)
+    x = abs(x)
+    y = abs(y)
+    z = abs(z)
+    m = max(x, y, z)
+    if isinf(m)
+        return m
+    elseif isfinite(3*m*m) && m*m != 0 # Scaling not necessary
+        return sqrt(abs2(x) + abs2(y) + abs2(z))
+    elseif m == 0
+        o = one(m)
+        return m*sqrt(abs2(x/o) + abs2(y/o) + abs2(z/o))
```

Comment:
As I said in [this comment](https://github.com/JuliaLang/julia/pull/15552#discussion_r56686106), all arguments are promoted to floating points.  See
```julia
julia> hypot(0, 0, 0)
0.0

julia> @code_warntype hypot(0, 0, 0)
Variables:
  #self#::Base.Math.#hypot
  x::Int64
  y::Int64
  z::Int64
  ####xs#7508#7524::Tuple{Float64,Float64}
  ######xs#7507#7509#7525::Tuple{Float64}
  ####xs#7510#7526::Tuple{Float64,Float64}
  ######xs#7507#7511#7527::Tuple{Float64}
  ####xs#7512#7528::Tuple{Float64,Float64}
  ######xs#7507#7513#7529::Tuple{Float64}

Body:
  begin  # math.jl, line 174:
      return (Base.Math.hypot)((Base.box)(Float64,(Base.sitofp)(Float64,x::Int64)),(Base.box)(Float64,(Base.sitofp)(Float64,y::Int64)),(Base.box)(Float64,(Base.sitofp)(Float64,z::Int64)))::Float64
  end::Float64
```